service-workers/service-worker/unregister-then-register.https.html

```diff
@@ -138,39 +78,29 @@
         });
   }, 'Unregister then register does not affect existing controllee');
 
-promise_test(function(t) {
-    var scope = 'resources/scope/resurrection';
-    var iframe;
-    var registration;
+promise_test(async function(t) {
+  const scope = 'resources/scope/resurrection';
+  const registration = await service_worker_unregister_and_register(t, worker_url, scope);
+  t.add_cleanup(() => registration.unregister());
 
-    return service_worker_unregister_and_register(t, worker_url, scope)
-      .then(function(r) {
-          t.add_cleanup(function() {
-              return service_worker_unregister(t, scope);
-            });
+  await wait_for_state(t, registration.installing, 'activated');
+  const iframe = await with_iframe(scope);
+  t.add_cleanup(() => iframe.remove());
 
-          registration = r;
-          return wait_for_state(t, r.installing, 'activated');
-        })
-      .then(function() {
-          return with_iframe(scope);
-        })
-      .then(function(frame) {
-          iframe = frame;
-          return registration.unregister();
-        })
-      .then(function() {
-          return navigator.serviceWorker.register(worker_url, { scope: scope });
-        })
-      .then(function() {
-          iframe.remove();
-          return with_iframe(scope);
-        })
-      .then(function(frame) {
-          assert_not_equals(
-              frame.contentWindow.navigator.serviceWorker.controller, null,
-              'document should have a controller');
-          frame.remove();
-        });
-  }, 'Unregister then register resurrects the registration');
+  await registration.unregister();
+  const newRegistration = await navigator.serviceWorker.register(worker_url, { scope });
+  t.add_cleanup(() => newRegistration.unregister());
+
+  iframe.remove();
+  const newIframe = await with_iframe(scope);
+  t.add_cleanup(() => newIframe.remove());
+
+  const iframeRegistration = await newIframe.contentWindow.navigator.serviceWorker.ready;
+
+  assert_true(!!iframeRegistration, 'iFrame has registration');
+  assert_true(!!newIframe.contentWindow.navigator.serviceWorker.controller, 'iFrame has controller');
+
+  assert_not_equals(registration, newRegistration, 'Registrations are different');
+  assert_equals(newRegistration, iframeRegistration, 'iFrame uses new registration');
```

Comment:
I don't think this holds because newRegistration and iframeRegistration are from different windows so have different prototypes.

Reply:
lol indeed. Will fix.